sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -349,9 +364,42 @@ internal class QProductCenterManager internal constructor(
         }
 
         purchasingCallbacks[purchaseModel.product.storeID] = callback
+
+        updatePurchaseOptions(purchaseModel.options, purchaseModel.product.storeID)
+
         billingService.purchase(context, purchaseModel)
     }
 
+    @VisibleForTesting
+    fun actualPurchaseOptions(): Map<String, QPurchaseOptions> {
+        processingPurchaseOptions?.let {
+            return it
+        }
+
+        val cachedPurchaseOptions = purchasesCache.loadProcessingPurchasesOptions()
+        processingPurchaseOptions = cachedPurchaseOptions
+
+        return cachedPurchaseOptions
+    }
+
+    private fun updatePurchaseOptions(options: QPurchaseOptions?, storeProductId: String?) {
+        storeProductId?.let { productId ->
+            val actualOptions = actualPurchaseOptions().toMutableMap()
+            options?.let {
+                actualOptions[productId] = it
+            } ?: run {
+                actualOptions.remove(productId)
+            }
+
+            processingPurchaseOptions = actualOptions.toMap()
```

Comment:
Why would we even recreate and reassign the map every time? Why don't we simply modify the existing one?